File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/Situation.java
package com.panopset.blackjackEngine;

/**
 * In Blackjack, there is a well defined set of situations. An instance of this class represents a
 * single situation. Each situation can be mapped to an action, based on a strategy.
 *
 */
public class Situation {

  final BlackjackCard dealerUpCard;
  final HandPlayer handPlayer;

  public Situation(final BlackjackCard dealerUpCard, final HandPlayer handPlayer) {
    this.dealerUpCard = dealerUpCard;
    this.handPlayer = handPlayer;
  }

  public HandPlayer getHandPlayer() {
    return handPlayer;
  }

  public BlackjackCard getDealerUpCard() {
    return dealerUpCard;
  }

  @Override
  public String toString() {
    return String.format("Dealer up: %s Hand: %s", dealerUpCard, handPlayer);
  }
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/Situation.kt
package com.panopset.blackjackEngine

/**
 * In Blackjack, there is a well defined set of situations. An instance of this class represents a
 * single situation. Each situation can be mapped to an action, based on a strategy.
 *
 */
class Situation(val dealerUpCard: BlackjackCard?, val handPlayer: HandPlayer?) {

    override fun toString(): String {
        return String.format("Dealer up: %s Hand: %s", dealerUpCard, handPlayer)
    }
}
